File: Tarea8/Libreria.java
import java.util.ArrayList;
import java.util.Scanner;

public class Libreria {

    private ArrayList<Usuario> usuarios;
    private ArrayList<Usuario> usuariosRenta;
    private ArrayList<Libro> listaLibros;
    private ArrayList<Libro> librosRentados;
    private ArrayList<Libro> librosNoRentados;
    boolean repetir = true;

    public Libreria() {
        this.usuarios = new ArrayList<>();
        this.listaLibros = new ArrayList<>();
        this.librosRentados = new ArrayList<>();
        this.usuariosRenta = new ArrayList<>();
        this.librosNoRentados = new ArrayList<>();
    }
    public void registrarUsuario(Usuario usuario) {
        this.usuarios.add(usuario);
    }
    public void añadirLibro(Libro libro) {
        this.listaLibros.add(libro);
    }
    public void librosRenta(Libro librosRenta) {
        this.librosRentados.add(librosRenta);
    }
    public void usuarioRenta(Usuario usuariosRenta) {
        this.usuariosRenta.add(usuariosRenta);
    }
    public void libroSinRenta(Libro librosNoRenta) {
        this.librosNoRentados.add(librosNoRenta);
    }

    public void mostrarUsuarios() {
        System.out.println("---USUARIOS---");
        for (Usuario usuario : this.usuarios) {
            System.out.println("-" + usuario.getNombre() + " ID: " +usuario.getId());
        }
    }
    public void mostrarLibros() {
        System.out.println("---LIBTROS---");
        for (Libro libro : this.listaLibros) {
            System.out.println("Datos del libro: ");
            System.out.println("Nombre del libro: " + libro.getName());
            System.out.println("Nombre del autor: " + libro.getAutor());
            System.out.println("ID del libro: " + libro.getId());
            System.out.println("Cantidad disponible: " + libro.getCantidad());
            System.out.println(" ");
        }
    }
    public void rentaLibros() {
        System.out.println("---Usuarios Registados---");
        for (int x = 0 ; x < usuarios.size() ; x++) {
            System.out.println(x + ". " + usuarios.get(x).getNombre());
        }
        System.out.println("Quien va rentar: ");
        Scanner scanner = new Scanner(System.in);
        int renta = scanner.nextInt();
        usuarioRenta(usuarios.get(renta));
        do {
            System.out.println("---Libros disponibles---");
            for (int y = 0; y < listaLibros.size(); y++) {
                System.out.println(y + ". " + listaLibros.get(y).getName());
            }
            System.out.println("Que libro desea rentar: ");
            int rentaLibro = scanner.nextInt();
            if (rentaLibro >= 0) {
                System.out.println("Se rento el libro: " + listaLibros.get(rentaLibro).getName());
                listaLibros.get(rentaLibro).rentarLibro();
                librosRenta(listaLibros.get(rentaLibro));
            }
            System.out.println("Quieres rentar mas libros? Y/N");
            char seguir = scanner.next().charAt(0);
            repetir = (seguir == 'Y' || seguir == 'y');
        }while(repetir);
        for (int x = 0 ; x < listaLibros.size() ; x++) {
            if(!librosRentados.contains(listaLibros.get(x))) {
                libroSinRenta(listaLibros.get(x));
            }
        }
    }
    public void mostrarRentadores() {
        System.out.println("Esta es la lista de rentadores");
        for (Usuario usuarioRenta : this.usuariosRenta) {
            System.out.println("-" + usuarioRenta.getNombre() + " ID: " + usuarioRenta.getId());
        }
    }
    public void mostrarLibrosRentados() {
        System.out.println("Estos son los libros que han sido rentados");
        for (Libro librosRentados : this.librosRentados) {
            System.out.println("-" + librosRentados.getName() + " By: " + librosRentados.getAutor() + " ID: " + librosRentados.getId());
            System.out.println("Cantidad restante: " + librosRentados.getCantidad());
        }
    }
    public void mostrarLibrosNoRentados() {
        System.out.println("Estos son los libros no rentados");
        for (Libro librosNoRentados : this.librosNoRentados) {
            System.out.println("-" + librosNoRentados.getName() + " By:" + librosNoRentados.getAutor() + " ID: " +librosNoRentados.getId());
            System.out.println("Cantidad restante: " + librosNoRentados.getCantidad());
        }
    }
}

File: Tarea8Python/Libreria.py
class Libreria:
    def __init__(self):
        self.usuarios = []
        self.lista_libros = []
        self.libros_rentados = []
        self.usuarios_renta = []
        self.libros_no_rentados = []

    def registrar_usuario(self, usuario):
        self.usuarios.append(usuario)

    def añadir_libro(self, libro):
        self.lista_libros.append(libro)

    def libros_renta(self, libros_renta):
        self.libros_rentados.append(libros_renta)

    def usuario_renta(self, usuarios_renta):
        self.usuarios_renta.append(usuarios_renta)

    def libro_sin_renta(self, libros_no_renta):
        self.libros_no_rentados.append(libros_no_renta)

    def mostrar_usuarios(self):
        print("---USUARIOS---")
        for usuario in self.usuarios:
            print("-" + usuario.get_nombre() + " ID: " + str(usuario.get_id()))

    def mostrar_libros(self):
        print("---LIBROS---")
        for libro in self.lista_libros:
            print("Datos del libro:")
            print("Nombre del libro: " + libro.get_name())
            print("Nombre del autor: " + libro.get_autor())
            print("ID del libro: " + str(libro.get_id()))
            print("Cantidad disponible: " + str(libro.get_cantidad()))
            print(" ")

    def renta_libros(self):
        print("---Usuarios Registrados---")
        for x, usuario in enumerate(self.usuarios):
            print(f"{x}. {usuario.get_nombre()}")

        print("Quién va a rentar: ")
        renta = int(input())
        self.usuario_renta(self.usuarios[renta])

        while True:
            print("---Libros disponibles---")
            for y, libro in enumerate(self.lista_libros):
                print(f"{y}. {libro.get_name()}")

            print("Qué libro desea rentar: ")
            renta_libro = int(input())
            if renta_libro >= 0:
                print("Se rentó el libro: " + self.lista_libros[renta_libro].get_name())
                self.lista_libros[renta_libro].rentar_libro()
                self.libros_renta(self.lista_libros[renta_libro])

            seguir = input("Quieres rentar más libros? (Y/N): ")
            if seguir.lower() != 'y':
                break

        for libro in self.lista_libros:
            if libro not in self.libros_rentados:
                self.libro_sin_renta(libro)

    def mostrar_rentadores(self):
        print("Esta es la lista de rentadores")
        for usuario_renta in self.usuarios_renta:
            print("-" + usuario_renta.get_nombre() + " ID: " + str(usuario_renta.get_id()))

    def mostrar_libros_rentados(self):
        print("Estos son los libros que han sido rentados")
        for libros_rentados in self.libros_rentados:
            print("-" + libros_rentados.get_name() + " By: " + libros_rentados.get_autor() + " ID: " + str(libros_rentados.get_id()))
            print("Cantidad restante: " + str(libros_rentados.get_cantidad()))

    def mostrar_libros_no_rentados(self):
        print("Estos son los libros no rentados")
        for libros_no_rentados in self.libros_no_rentados:
            print("-" + libros_no_rentados.get_name() + " By: " + libros_no_rentados.get_autor() + " ID: " + str(libros_no_rentados.get_id()))
            print("Cantidad restante: " + str(libros_no_rentados.get_cantidad()))
